# HG changeset patch
# Parent e4f53aeb83c3563ad0e75fc67c1f11fcb4d284b9

Properly handle kick for a channel member already in a zombie state.

Backported from snircd 1.4.0 - http://hg.quakenet.org/snircd/rev/5f1306ec2068

diff -r e4f53aeb83c3 ircd/m_kick.c
--- a/ircd/m_kick.c	Sat Jun 29 16:18:12 2013 +0200
+++ b/ircd/m_kick.c	Sun Jun 30 14:55:19 2013 +0200
@@ -214,8 +214,16 @@
     return 0;
 
   /* We go ahead and pass on the KICK for users not on the channel */
-  if (!(member = find_member_link(chptr, who)) || IsZombie(member))
+  if (!(member = find_member_link(chptr, who)) || IsZombie(member)) {
+    /* 
+     * target user is a zombie
+     *   make_zombie will figure out if the user should remain a zombie
+     *   or be removed from the channel now
+     */
+    if (member)
+      make_zombie(member, who, cptr, sptr, chptr);
     member = 0;
+  }
 
   /* Send HACK notice, but not for servers in BURST */
   /* 2002-10-17: Don't send HACK if the users local server is kicking them */
